src/tests/class_tests/openms/source/Tagger_test.cpp

```diff
@@ -0,0 +1,262 @@
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "TESTT") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "ESTTH") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "STTHI") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "TTHIS") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "THIST") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "HISTA") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "ISTAG") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "STAGG") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "TAGGE") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "AGGER") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "GGER") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "GER") != tags.end(), true)
+
+  // tagger searching for charge +2 with fixed modification
+  Tagger tagger4 = Tagger(2, 10, 5, 2, 2, ListUtils::create<String>("Oxidation (D)"));
+  tags.clear();
+  tagger4.getTag(spec2, tags);
+  TEST_EQUAL(tags.size(), 667);
+
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "PEPT") != tags.end(), false)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "PEPTI") != tags.end(), false)
+  // modified residue found again
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "EPTID") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "PTIDE") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "TIDET") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "IDETE") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "DETES") != tags.end(), true)
+
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "ETEST") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "TESTT") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "ESTTH") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "STTHI") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "TTHIS") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "THIST") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "HISTA") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "ISTAG") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "STAGG") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "TAGGE") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "AGGER") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "GGER") != tags.end(), true)
+  TEST_EQUAL(std::find(tags.begin(), tags.end(), "GER") != tags.end(), true)
+
+  // tagger searching for charge +2 with variable modification
+  Tagger tagger5 = Tagger(2, 10, 5, 2, 2, StringList(), ListUtils::create<String>("Oxidation (D)"));
```

Comment:
would it make sense to distinguish "unmodified" tags as well as 
modified ones? There are probably several use cases where one might 
want to distinguish the Tag: PEPD from PEPD(Oxidiation)

Reply:
This would be easy to do, I would need Strings for each residue, e.g. "D" and "D(Oxidation)" and StringLists for the list of strings. The filtering would also change from 2 or 3 letters to matching strings of an undefined length. I don't think I can squeeze both of them into the same class without copying much of the code

Reply:
yeah probably one would then use Residue* instead of char to be fully generic.